src/commands/messageContent/AutoReact.ts

```diff
@@ -0,0 +1,39 @@
+import { Message } from "discord.js";
+
+import { replaceMentions } from "../../utils/helpers";
+import { discordBot } from "../../server";
+
+const reactionMap = new Map<string | RegExp, string>([
+    [/hodor/i, "door"],
+    [/bisbopt/i, "bisbopt"],
+    [/duck/i, "duck"],
+    [/69/i, "nice"],
+    [/420/i, "weed"],
+    [/puya[1ilӏ]{1,2}up/i, "downvote"],
+    [/bruh/i, "bruh"]
+]);
+
+export default {
+    trigger: /(hodor|bisbopt|duck|69|420|puya[1ilӏ]{1,2}up|bruh)/i,
+    handler: (message: Message) => {
+        message.content = replaceMentions(message);
+        for (const [ trigger, reaction ] of reactionMap) {
+            if ((trigger instanceof RegExp)) {
+                if (!trigger.test(message.content)) {
+                    continue;
+                }
+            } else {
+                if (!message.content.toLowerCase().includes(trigger)) {
```

Comment:
One thing to note - gifs / links / etc are technically text content messages to the client, so to avoid reacting `nice` to every link with a `69` in it, for example, I've been using a [custom string matcher](https://github.com/reddit-seattle/seabot/blob/7fd89b34a1a060bfc58dd1e6a03437fa8d62feb3/src/utils/reaccs.ts#L60) that accounts for the following options for auto-react features:
- trimming whitespace so `partial` can match "part i al"
- replacing URL strings with `''` to not auto-react to text in links or gifs

Reply:
Somehow I missed the custom string matcher; I've re-implemented the functionality with some minor differences.